src/modules/tracer.js

```diff
@@ -0,0 +1,63 @@
+const path = require('path');
+
+/**
+ * Traces the caller of the logtify module
+ */
+class Tracer {
+  /**
+   * Try to get the information about caller module
+   * function, that called the module, the name of the file this function is from and the name of the project
+   * This data will be used in prefixes
+   * @return {object} - {
+   *   function: {string}, - name of the callee function. 'anonymous' if none or arrow function
+   *   module: {string}, - name of the parent module
+   *   project: {string} - name of the parent module. Empty if none
+   * }
+   */
+  trace() {
+    try {
+      const callerFileInfo = this.getCallerFileInfo();
+      const callerFunction = callerFileInfo.getFunctionName();
+      const callerModule = callerFileInfo.getFileName().split(path.sep).pop();
+      // if logtify was required using absolute local path or during tests, there is no parent project
+      // So we check if the __dirname has node_modules inside to make sure it was called as a dependency
+      // So if we have __dirname as /var/lib/MyAwesomeProject/node_modules/node-logtify/...
+      // We eventually get MyAwesomeProject
+      const callerProject = __dirname.includes('node_modules') ?
+        __dirname.split(`${path.sep}node_modules${path.sep}`)[0].split(path.sep).pop() : '';
+      return {
+        function: callerFunction || 'anonymous',
+        module: callerModule,
+        project: callerProject
+      };
+    } catch (e) {
+      return {};
```

Comment:
Error's stack is used to retrieve the caller module. I wrapped this in try{} catch{} for sanity that in case of some specific environment/structure/usage that I was not aware of, we will not end up with an error but just get some default value. A good example is when a module is called from the root of the module and does not have a function. Or if a function is anonymous.